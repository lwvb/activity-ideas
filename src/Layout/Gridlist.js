import React, { Component } from 'react';
import {Grid, Row, Col} from 'react-flexbox-grid/lib';

class Gridlist extends Component {
    render() {
        return (
            <Grid fluid className="">
                <Row>
                    {React.Children.map(this.props.children, (child) => 
                        <Col xs={12} sm={6} md={4} lg={3}>{child}</Col>
                    )}
                </Row>
            </Grid>
        );
    }
}

export default Gridlist;