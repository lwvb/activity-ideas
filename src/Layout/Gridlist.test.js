import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils';
import TestRenderer from 'react-test-renderer';

import Gridlist from './Gridlist';
  

describe('Gridlist component', () => {
    it('renders', () => {
        const component = TestRenderer.create(<Gridlist />);
        expect(component.toJSON()).toMatchSnapshot()
    });

    it('outputs an empty grid without children', () => {
        const component = ReactTestUtils.renderIntoDocument(<Gridlist />);
        const rows = ReactTestUtils.scryRenderedDOMComponentsWithClass(component, 'row').length
        const cols = ReactTestUtils.scryRenderedDOMComponentsWithClass(component, 'col').length
        expect(rows).toBe(1);
        expect(cols).toBe(0);
    });

    it('wraps children in columns', () => {
        const component = ReactTestUtils.renderIntoDocument(<Gridlist><div>Child1</div><div>Child2</div><div>Child3</div></Gridlist>);
        const rows = ReactTestUtils.scryRenderedDOMComponentsWithClass(component, 'row').length
        const cols = ReactTestUtils.scryRenderedDOMComponentsWithClass(component, 'col-xs-12').length
        expect(rows).toBe(1);
        expect(cols).toBe(3);
    })
     
});
