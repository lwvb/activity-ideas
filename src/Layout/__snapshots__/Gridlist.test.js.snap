// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gridlist component renders 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  />
</div>
`;
